packages/origin-backend-app/src/notification/handlers/certificate-approved.handler.ts

```diff
@@ -0,0 +1,76 @@
+import { EventsHandler, IEventHandler, QueryBus } from '@nestjs/cqrs';
+import { Logger } from '@nestjs/common';
+import { ModuleRef } from '@nestjs/core';
+import { DeviceService, UserService } from '@energyweb/origin-backend';
+import { Role } from '@energyweb/origin-backend-core';
+import { CertificateCreatedEvent, GetCertificationRequestQuery } from '@energyweb/issuer-api';
+import { RegistrationService } from '@energyweb/origin-organization-irec-api';
+import { MailService } from '../../mail';
+
+@EventsHandler(CertificateCreatedEvent)
+export class CertificateApprovedHandler implements IEventHandler<CertificateCreatedEvent> {
+    private readonly logger = new Logger(CertificateApprovedHandler.name);
+
+    private _deviceService: DeviceService;
+
+    constructor(
+        private readonly mailService: MailService,
+        private readonly userService: UserService,
+        private readonly registrationService: RegistrationService,
+        private readonly queryBus: QueryBus,
+        private readonly moduleRef: ModuleRef
+    ) {}
+
+    private get deviceService() {
+        if (this._deviceService) {
+            return this._deviceService;
+        }
+
+        this._deviceService = this.moduleRef.get<DeviceService>(DeviceService, {
+            strict: false
+        });
+
+        return this._deviceService;
+    }
```

Comment:
@kosecki123 @JosephBagaric 
I do not like this, but I tried several other ways and they do not work. Maybe you have ideas how to inject `DeviceService`?

Reply:
The problem is that the DeviceModule is currently a dynamic module that messes up with the dependency injection. For now I do know any better solution.